server/sql_info_fetcher.go

```diff
@@ -255,14 +273,8 @@ func (sh *sqlInfoFetcher) getExplainAnalyze(ctx context.Context, sql string, res
 	resultChan <- &explainAnalyzeResult{rows: rows}
 }
 
-func (sh *sqlInfoFetcher) catchCPUProfile(ctx context.Context, sec int, zw *zip.Writer, errChan chan<- error) {
-	// dump profile
-	fw, err := zw.Create("profile")
```

Comment:
Call `zw.Create("profile")` in  `dumpCPUProfile()` instead of `catchCPUProfile()`,
For avoiding data race with 
https://github.com/pingcap/tidb/blob/5d76de903a6d2c95f83da365e5d757dad9fb9e7b/server/sql_info_fetcher.go#L207